src/IssueViz.Security/Hotspots/HotspotMatcher.cs

```diff
@@ -30,11 +32,24 @@ internal interface IHotspotMatcher
     }
 
     [Export(typeof(IHotspotMatcher))]
-    internal class NoOpHotspotMatcher : IHotspotMatcher
+    internal class HotspotMatcher : IHotspotMatcher
     {
         public bool IsMatch(IAnalysisIssueVisualization localHotspotVisualization, SonarQubeHotspot serverHotspot)
         {
-            return false;
+            if (!StringComparer.OrdinalIgnoreCase.Equals(localHotspotVisualization.RuleId, serverHotspot.Rule.RuleKey)
+                || !PathHelper.IsServerFileMatch(localHotspotVisualization.FilePath, serverHotspot.FilePath))
+            {
+                return false;
+            }
+
+            if (!string.IsNullOrEmpty(serverHotspot.LineHash)
+                && StringComparer.Ordinal.Equals(localHotspotVisualization.LineHash, serverHotspot.LineHash))
+            {
+                return true;
+            }
+
+            return localHotspotVisualization.StartLine == serverHotspot.TextRange.StartLine 
+                   || StringComparer.Ordinal.Equals(localHotspotVisualization.Issue.PrimaryLocation.Message, serverHotspot.Message);
```

Comment:
The issue matching for non-hotspots doesn't look at the 'Message', does it?

What's the reason for using a different algorithm here?

Using a different algorithm is fine as long as there is a reason for it. However, we need to capture the reason for it, otherwise it will look like a bug to anyone looking at the code in the future.

Reply:
I got this idea from talking to Damien. Since there's no hashes for hotspots in the API response, they use message as an additional feature to match based on it.

Reply:
Ok, makes sense (although that does beg the question that if we don't have hash for hotspots, why is the code above checking for them at all?).

This is a clear design/implementation decision that deserves to be captured. Otherwise, it's not obvious whether it's a mistake or not e.g.


```
// We don't currently expect to have hashes for hotspopts so we use the same approach as other SL flavours and use the exact message as an extra value to match on (we don't check the message for non-hotspot issues since we do have hashes).
```